controllers/fio/controller.go

```diff
@@ -70,6 +75,10 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// The cr could have been modified since the last time we got it
```

Comment:
I'm not quite sure what scenario this Client.Get covers.
One thought regarding the comment: The jobs should be immutable, as handling modification would be really cumbersome: stopping and rerunning, cleaning etc. 

WDYT?

Reply:
> I'm not quite sure what scenario this Client.Get covers.
Scenario: the controller sets cr.Status.Running to true, starts the job, the job finishes, because it's very fast (<1s for a fast fio job), still in the same Reconcile function the controller checks if the job is finished, it is, so it tries to set cr.Status.Running to false and Completed to true. Boom. Conflict, because the cr's resourceVersion was updated in the first Running update. This Get solves this issue by making sure we will try to update the status of the latest CR.

> One thought regarding the comment: The jobs should be immutable, as handling modification would be really cumbersome: stopping and rerunning, cleaning etc.
We're not rerunning / modifying anything. This comment only applies to the Status of the CR, has nothing to do with any jobs.

Reply:
thanks for the clarification. 
food for thought: we should think about serializing the reconcile loop to resolve such situations in the future.